### ISSUE TITLE
Repeat: multi-stream mode

### ISSUE THREAD

**Issue by marcusmueller** [gnuradio/gnuradio#5113]
As [discussed](https://lists.gnu.org/archive/html/discuss-gnuradio/2021-09/msg00117.html) on the mailing list:

It would be good if the Repeat block supported multiple streams. To do would be:

* change the `io_signature` in `repeat_impl.cc` to allow from 1 to -1 (==infinity) in- and outputs
* add a **last** `unsigned int number_of_streams` parameter to the constructor and to `make` (watch the headers!) that defaults to 1
* add a `check_topology` to the same file to check that the number of in- and outputs is equal
* add another `for` loop to the `work` method to iterate over all `input_items[0…N]` / `output_items[0…N]`
* in `blocks_repeat.block.yml` add the `number_of_streams` parameter and use it to set the `multiplicity` of the `inputs:` and `outputs:` element (compare `blocks_add_xx.block.yml`)

**Comment by me-hem** [comment]
@marcusmueller , could you please assign this issue to me as part of Hactoberfest?

**Comment by marcusmueller** [comment]
@me-hem I mean, we're a reliable FOSS project, where code reviews take effort: No, we do not _assign_ issues to fresh contributors from the internet that explicitly are here to get a t-shirt!

The process is that you open a PR that you have, yourself, written, compiled, tested well and that solves the issue :) Then, we merge it :) We only _assign issues_ to known long-term contributors in cases where they themselves can estimate the effort taken.

So, I'm quite looking forward to your first PR :) any of these open issues are here for you to tackle right away, no need for me to assign things to anyone!

**Comment by me-hem** [comment]
@marcusmueller I agree with your point, but I am not here for just a T-shirt. I am part of GSSoC, and asking project maintainers to get issues assigned is like a protocol in all the projects I contributed to (also big ones like KubeArmor).
  
Thank you for your comment.

**Comment by marcusmueller** [comment]
We are a GSoC organization ourselves. We do not require this. Anyways, still looking forward to your first PR.

**Comment by me-hem** [comment]
@marcusmueller Yeah sure.
